### PR TITLE
represent numbers as strings as they are now decimal128 which we cann…

### PR DESCRIPTION
**Pull request by zoren**
…ot represent exactly as number

### CODE REVIEW

File: lib/DeonData.ts [hunk 1]
```diff
@@ -192,9 +192,9 @@ export const mkIntValue = (n: number): IntValue =>
 
 export interface FloatValue {
   class: 'FloatValue';
-  d: number;
+  d: string;
 }
-export const mkFloatValue = (n: number): FloatValue =>
+export const mkFloatValue = (n: string): FloatValue =>
```
Comment (adamschoenemann) [inline lib/DeonData.ts]: Maybe we should do some validation of the string here? Throw an exception upon creation if it is not a valid number

Comment (adamschoenemann) [reply lib/DeonData.ts]: Something ala https://stackoverflow.com/questions/175739/built-in-way-in-javascript-to-check-if-a-string-is-a-valid-number perhaps

Comment (zoren) [reply lib/DeonData.ts]: So `let isNumeric = s => s !== '' && !isNaN(s)`? In that case I think we should also check that the parameter in `mkIntValue` is `int32`.

Comment (adamschoenemann) [reply lib/DeonData.ts]: Fine by me :+1:

Comment (zoren) [reply lib/DeonData.ts]: Actually we should also check if it's finite. And that is kinda difficult without parsing the number ourselves. `isFinite(+"1e309") === false` but the range for decimal128 is all the way up to `1e6144`. So I don't think we should do this.

**Comment by hansbugge** [comment]
You should make an entry in the changelog.

**Comment by bjornbugge** [comment]
Whoops -- too fast!  Sorry @hansbugge

**Comment by zoren** [comment]
@hansbugge @bjornbugge https://github.com/deondigital/api-client/pull/73

**Pull request merged**